packages/admin/cms-admin/src/dam/MoveDamItemDialog/MoveDamItemDialog.tsx

```diff
@@ -113,25 +195,34 @@ export const MoveDamItemDialog = ({ onClose, onChooseFolder, numberOfItems }: Mo
                 >
                     <FormattedMessage id="comet.dam.moveDamItemDialog.startOver" defaultMessage="Start over" />
                 </Button>
-                <Button
+                <SaveButton
                     startIcon={<Move />}
                     variant="contained"
-                    onClick={() => {
-                        if (selectedId !== undefined) {
-                            onChooseFolder(selectedId);
-                        }
+                    onClick={async () => {
+                        await moveSelected();
+                        onClose();
                     }}
                     disabled={selectedId === undefined}
+                    saving={moving}
+                    hasErrors={hasErrors}
                 >
                     <FormattedMessage
                         id="comet.dam.moveDamItemDialog.moveItems"
                         defaultMessage="Move {num, plural, one {item} other {items}}"
                         values={{
-                            num: numberOfItems,
+                            num: damItemsToMove.length,
                         }}
                     />
-                </Button>
+                </SaveButton>
             </DialogActions>
         </FixedHeightDialog>
     );
 };
+
+export const MoveDamItemDialog = (props: MoveDamItemDialogProps) => {
+    if (!props.open) {
```

Comment:
Why do you need this additional component? Because of the early return and hooks? Wouldn't it be possible to define the hooks above the early return?

Reply:
Without this component, the visibility of the dialog is controlled via `open` but the component is never dismounted. So the states are preserved which may not be desired between uses of the `MoveDamItemDialog`.

My easy workaround was to wrap the component and return null when open is false, causing the inner component to dismount. An alternative would be to use a `useEffect()` to reset all the states if open is set to false. Or to control the rendering with a condition in the consumer component (which was the previous solution but goes against the Mui patterns).
Do you have another idea?